[David Au] iP

dvdweien opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

dvdweien committed:
Level 1. Greet, Echo, Exit

dvdweien committed:
Level 2. Add, List

Add the ability to store whatever text entered by the user and
display them back to the user when requested.

dvdweien committed:
Level-3: Mark as Done

Add the ability to mark tasks as done.

dvdweien committed:
Level-4: ToDo, Event, Deadline

Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it e.g., visit new
theme park
Deadlines: tasks that need to be done before a specific date/time
e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific time and ends at a
specific time e.g., team project meeting on 2/10/2019 2-4pm

dvdweien committed:
A-TextUiTesting: Automated Text UI Testing

Use the input/output redirection technique to semi-automate the
testing of Duke.

dvdweien committed:
Level-5: Handle Errors

Teach Duke to deal with errors such as incorrect inputs entered
by the user.

dvdweien committed:
Level-6: Delete

Add support for deleting tasks from the list.

dvdweien committed:
Add ability to save

Program is unable to save.

The ability to save is vitally important for a user to be able to
remember their tasks.

Let's add the ability to save

dvdweien committed:
Make Storage class do processing of data

When loading the data, the Storage class does not do any
processing of the data and simply passes the file

Processing of the data is a job that should be done by the
Storage class

Have the Storage class process the data before passing it over

dvdweien committed:
Add ability to understand dates

Program dates are simply stored as strings with no ability for
manipulation

Understanding dates is important for functionality such as
listing tasks on a particular day

Dates are now parsed as LocalDate objects

dvdweien committed:
Merge branch 'branch-Level-8' into branch-Level-7

dvdweien committed:
Fix bugs after merging

Program has bugs due to the merge

Bugs must be fixed for the program to work

The bugs are fixed

dvdweien committed:
Merge branch 'master' of https://github.com/dvdweien/ip

dvdweien committed:
Make the code more OOP

Roles of classes are not well defined

Code is not OOP enough

Refactor the code to extract out closely related code as classes.

dvdweien committed:
Divide classes into packages

Classes are not packaged

Classes are unorganized

Organize classes by putting them into packages

dvdweien committed:
Add JUnit tests to test the behavior of the code.

Testing is being done manually.

Automating tests allows for more consistency and speed up the
testing workflow

Tests have been implemented with JUnit

dvdweien committed:
Fix bug when packaged in Jar file

Program does not run from Jar file due to a bug

Get the program to run from jar

Fix the bug

dvdweien committed:
Add JavaDocs

No JavaDocs

Not having JavaDocs makes it difficult for other programmers to
understand the code

JavaDocs are added

dvdweien committed:
Tweak the code to comply with the coding standard

Code does not comply with the coding standard

Code does not look consistent

Tweak the code to comply with the coding standard

dvdweien committed:
Find a task by searching for a keyword

Program does not support finding tasks relevant to a keyword

Tedious to read through all the tasks in the list to find related
tasks

Added functionality to search through list for tasks with the
keyword

dvdweien committed:
Merge branch 'branch-A-JavaDoc'

dvdweien committed:
Merge branch 'branch-Level-9'

dvdweien committed:
Merge branch 'branch-A-CodingStandard'

dvdweien committed:
Merge branch 'add-gradle-support' of https://github.com/dvdweien/ip

dvdweien committed:
Use Gradle

project does not use Gradle

Automate project builds using Gradle

Project uses Gradle

samuelfangjw commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+import java.time.format.DateTimeFormatter;
+
+class Deadline extends Task {
+	private final LocalDate time;
```
Perhaps this should be 4 spaces rather than a tab? I noticed you used tabs for indentation in some of your other classes too.

samuelfangjw commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+	 * Deadline constructor where the boolean isDone is set to false
+	 * @param a name of Deadline
+	 * @param b date of Deadline in yyyy-mm-dd (e.g. 2021-01-31)
+	 */
```
Perhaps there should be an empty line between description and parameter sections, as well as punctuation after each parameter description? I noticed this in your other javadoc comments too.

samuelfangjw commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,24 @@
+
+import java.util.Scanner;
+
+public class Duke {
```
Perhaps there should be a javadoc comment here?

samuelfangjw commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,24 @@
+
+        String input = sc.nextLine();
+
+        while(!input.equals("bye")) {
```
Perhaps while should be followed by a whitespace character?

samuelfangjw commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,24 @@
+        	input = sc.nextLine();
+        }
+
+        System.out.println("    ____________________________________________________________");
+		System.out.println();
+		System.out.println("      Bye. Hope to see you again soon!");
+		System.out.println("    ____________________________________________________________");
+		System.out.println();
```
Perhaps these lines should all be indented at the same level?

samuelfangjw commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,119 @@
+import java.util.Scanner;
+import java.util.function.Consumer;
+
+public class Parser {
```
Perhaps there should be a javadoc since this is a public class? I noticed the same issue in some of your other classes too.

samuelfangjw commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+
+class TaskList {
+	
+	private final ArrayList<Task> list;
```
Perhaps list should be in plural form?

samuelfangjw commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+	}
+
+	/**
+	 * get the entire Task ArrayList
```
Perhaps the first word of the javadoc should be Returns?

samuelfangjw commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,129 @@
+	/**
+	 * displays the message when an invalid input is given
+	 */
+	static void invalidInput() {
```
Perhaps the name of the method should be a verb instead? I noticed some of your other methods have the same issue.

samuelfangjw reviewed (commented):
Overall, I found your code easy to read. For the indentation issues, I suggest setting your IDE to automatically convert tabs to spaces.

dvdweien committed:
Add a GUI to Duke

Duke runs on the command line

Command line is not user-friendly

Use the JavaFX technology to implement a GUI

dvdweien committed:
Use CheckStyle

Coding violations have to be checked manually

Checkstyle can help spot coding violations automatically

Prabhakaran-Gokul commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+     * @return true only if Deadline falls on the given day and !isDone
+     */
+    @Override
+    boolean onDay(String s) {
```
Perhaps it is better to name this method as isOnDay?

Prabhakaran-Gokul commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+     * @param b date of Deadline in yyyy-mm-dd (e.g. 2021-01-31)
+     * @param c value that boolean isDone will be set to
+     */
+    Deadline(String a, String b, boolean c) {
```
Perhaps you can consider choosing a more descriptive name for your parameters?

Prabhakaran-Gokul commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+    private boolean isExited;
+
+    /**
+     * loads the txt file into mem and displays the welcome message
```
Maybe the starting letter can be capitalised?

Prabhakaran-Gokul reviewed (commented):
Just a few fixes but LGTM! Well done and all the best for your next increments!

dvdweien committed:
Fix jar bug after adding JavaFX

JavaFX introduced some bugs when creating the JAR file

Fix the bugs

So that the JAR file can work

dvdweien committed:
Use Assertions

Use assert feature (not JUnit assertions) to document important
assumptions that should hold at various points in the code.

dvdweien committed:
Merge pull request #1 from dvdweien/branch-A-Assertions

Use Assertions

dvdweien committed:
Merge branch 'master' of https://github.com/dvdweien/ip

dvdweien committed:
Improve code quality

Critically examines the code and refactor to improve the code
quality where necessary.

dvdweien committed:
Merge pull request #2 from dvdweien/branch-A-CodeQuality

Branch a code quality

dvdweien committed:
Merge branch 'master' of https://github.com/dvdweien/ip

dvdweien committed:
More flexibility in searching for items

All more flexibility in search e.g., find items even if the
keyword matches the item only partially.

dvdweien committed:
A-BetterGui

Improve the GUI to make it more polished

dvdweien committed:
A-UserGuide

Add a User Guide to the project.

dvdweien committed:
Set theme jekyll-theme-slate
